pkg/controller/controller_test.go

```diff
@@ -2781,3 +2747,192 @@ func TestProvisionFromPVC(t *testing.T) {
+				ReclaimPolicy: &deletePolicy,
+			},
+			PVName: "test-name",
+			PVC: &v1.PersistentVolumeClaim{
+				ObjectMeta: metav1.ObjectMeta{
+					UID:         "testid",
+					Name:        "fake-pvc",
+					Annotations: tc.annotation,
+				},
+				Spec: v1.PersistentVolumeClaimSpec{
+					Selector: nil, // Provisioner doesn't support selector
+					Resources: v1.ResourceRequirements{
+						Requests: v1.ResourceList{
+							v1.ResourceName(v1.ResourceStorage): resource.MustParse(strconv.FormatInt(requestBytes, 10)),
+						},
+					},
+				},
+			},
+		}
+
+		_, state, err := csiProvisioner.(controller.ProvisionerExt).ProvisionExt(volOpts)
+		if tc.expectErr && err == nil {
+			t.Errorf("test %q: Expected error, got none", k)
+		}
+		if !tc.expectErr && err != nil {
+			t.Errorf("test %q: got error: %v", k, err)
+		}
+
+		if controller.ProvisioningFinished != state {
+			t.Errorf("test %q: expected ProvisioningState %s, got %s", k, controller.ProvisioningFinished, state)
+		}
+
+	}
+}
+
+func TestDeleteMigration(t *testing.T) {
+	testCases := []struct {
+		name           string
+		pv             *v1.PersistentVolume
+		translateToCSI bool
+		expectErr      bool
+	}{
+		{
+			name: "normal migration",
+			// The PV could be any random in-tree plugin - it doesn't really
+			// matter here. We only care that the translation is called and the
+			// function will work after some CSI volume is created
+			pv:             &v1.PersistentVolume{},
+			translateToCSI: true,
+		},
+		{
+			name: "no migration",
+			pv: &v1.PersistentVolume{
+				Spec: v1.PersistentVolumeSpec{
+					PersistentVolumeSource: v1.PersistentVolumeSource{
+						CSI: &v1.CSIPersistentVolumeSource{
+							VolumeHandle: "mock-id",
+						},
+					},
+				},
+			},
+			translateToCSI: false,
+		},
+	}
+	for _, tc := range testCases {
+
+		// Set up test
+		tmpdir := tempDir(t)
+		defer os.RemoveAll(tmpdir)
+		mockController, driver, _, controllerServer, csiConn, err := createMockServer(t, tmpdir)
+		if err != nil {
+			t.Fatal(err)
+		}
+		mockTranslator := NewMockProvisionerCSITranslator(mockController)
+		defer mockController.Finish()
+		defer driver.Stop()
+		clientSet := fakeclientset.NewSimpleClientset()
+		pluginCaps, controllerCaps := provisionCapabilities()
+		csiProvisioner := NewCSIProvisioner(clientSet, 5*time.Second, "test-provisioner", "test", 5, csiConn.conn, nil, driverName, pluginCaps, controllerCaps, "", false, mockTranslator)
+
+		// Set expectations
+		if tc.translateToCSI {
+			mockTranslator.EXPECT().IsPVMigratable(gomock.Any()).Return(true).Times(1)
+			mockTranslator.EXPECT().TranslateInTreePVToCSI(gomock.Any()).Return(
+				&v1.PersistentVolume{
+					Spec: v1.PersistentVolumeSpec{
+						PersistentVolumeSource: v1.PersistentVolumeSource{
+							CSI: &v1.CSIPersistentVolumeSource{
+								VolumeHandle: "mock-id",
+							},
+						},
+					},
+				}, nil).Times(1)
+		} else {
+			mockTranslator.EXPECT().IsPVMigratable(gomock.Any()).Return(false).AnyTimes()
```

Comment:
When `tc.translateToCSI` is false, we expect one call to `IsPVMigratable`. Given that, will it make sense to have `.Times(1)` here (instead of `.AnyTimes()`)?